extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -139,10 +151,35 @@ public NettyHttpInboundChannelAdapter(int port) {
 	}
 
 	public NettyHttpInboundChannelAdapter(int port, boolean ssl) {
+		/* begin whu */
+		this(port, "/", ssl);
+		/* end whu */
+	}
+
+	/* begin whu */
+	public NettyHttpInboundChannelAdapter(int port, String path) {
+		this(port, path, false);
+	}
+
+	public NettyHttpInboundChannelAdapter(int port, String path, boolean ssl) {
 		this.port = port;
 		this.ssl = ssl;
+		this.path = path;
+		putToChannelAdapter();
 	}
 
+	private void putToChannelAdapter() {
+		Map<String, Object> pathToProducer = getModuleState();
+		if (pathToProducer.containsKey(path)) {
+			String msg = "Channel for HTTP port " + port + ", path '" + path + "' already exists.";
+			logger.warn(msg);
+			throw new RuntimeException(msg);
+		}
+		pathToProducer.put(path, this);
```

Comment:
It would be better to store the `Method` instead of `this` - avoids the reflection to look up the method on every request.